src/controller/AutoCommissioner.cpp

```diff
@@ -102,20 +102,27 @@ CommissioningStage AutoCommissioner::GetNextCommissioningStage(CommissioningStag
     {
         return CommissioningStage::kCleanup;
     }
+
     switch (currentStage)
     {
     case CommissioningStage::kSecurePairing:
-        return CommissioningStage::kArmFailsafe;
-    case CommissioningStage::kArmFailsafe:
+        return CommissioningStage::kReadVendorId;
+    case CommissioningStage::kReadVendorId:
+        return CommissioningStage::kReadProductId;
+    case CommissioningStage::kReadProductId:
+        return CommissioningStage::kReadSoftwareVersion;
+    case CommissioningStage::kReadSoftwareVersion:
         if (mNeedsNetworkSetup)
         {
             return CommissioningStage::kGetNetworkTechnology;
         }
         else
         {
-            return CommissioningStage::kConfigRegulatory;
+            return CommissioningStage::kArmFailsafe;
         }
     case CommissioningStage::kGetNetworkTechnology:
+        return CommissioningStage::kArmFailsafe;
+    case CommissioningStage::kArmFailsafe:
```

Comment:
I had not thought about this before.  I suppose the fail-safe must be armed before any writes, but is there any advantage to setting it earlier anyway?  There is one additional piece of state that the fail-safe can revert, and that's the open PASE connection itself.

Is there a reason then to still arm it as a first step after establishing PASE, even though we start with a bunch of attribute reads?  I'm not necessarily advocating for this.  I'm just asking the question.

Reply:
I don't think it matters here, TBH. If the reads fail, the arm failsafe will likely also fail. The failsafe is meant to protect the device side from making changes if commissioning doesn't complete. Since reads don't really affect any thing, I didn't protect them.